src/daft-distributed/src/pipeline_node/scan_source.rs

```diff
@@ -148,8 +219,31 @@ impl DistributedPipelineNode for ScanSourceNode {
         stage_context: &mut StageExecutionContext,
     ) -> SubmittableTaskStream {
         let (result_tx, result_rx) = create_channel(1);
-        let execution_loop = self.execution_loop(result_tx, stage_context.task_id_counter());
-        stage_context.spawn(execution_loop);
+
+        // Check if this is a map-only pipeline by examining the stage type
+        // (this is performed at construction time from the LogicalPlanRef)
+        // And make sure that we only have 1 scan task
+        if self.is_map_only_pipeline && self.scan_tasks.len() == 1 {
```

Comment:
I tried to get this to be:
```rust
let execution_fut = if .... else ... ;
stage_context.spawn(execution_fut);
```

But I can't get the types to align. It's really an `impl Future<Output=Result<(), DaftError>>` but we cannot use `impl` for type annotations for variables. And unfortunately `rustc` gets tripped up and overly specific when it sees the `async move { ... }` function. It can't resolve this specific type with the `impl Future` of `self.execution_loop.